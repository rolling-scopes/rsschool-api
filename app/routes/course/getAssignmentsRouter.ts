import { INTERNAL_SERVER_ERROR, OK } from 'http-status-codes';
import * as Router from 'koa-router';
import { STATES, connection } from 'mongoose';
import { IApiResponse } from '../../models';
import { AssignmentsModel, IAssignmentsModel } from '../../models/assignments';

export const getAssignments = async (ctx: Router.IRouterContext) => {
    const userId = ctx.state.user!._id;
    if (connection.readyState !== STATES.connected) {
        ctx.status = INTERNAL_SERVER_ERROR;
        return;
    }
    try {
        const tasks = await AssignmentsModel.find({
            studentId: userId,
        }).exec();
        const body: IApiResponse<IAssignmentsModel> = {
            data: tasks,
        };
        ctx.body = body;
        ctx.status = OK;
    } catch (err) {
        ctx.logger.error(err);
        ctx.status = INTERNAL_SERVER_ERROR;
    }
};
