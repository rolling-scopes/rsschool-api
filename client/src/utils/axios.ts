import { AxiosRequestConfig } from 'axios';
import { NextPageContext } from 'next';
import getConfig from 'next/config';

const { serverRuntimeConfig } = getConfig();

export function getServerAxiosProps(ctx?: NextPageContext): Partial<AxiosRequestConfig> {
  return {
    baseURL: serverRuntimeConfig.rsHost || '',
    headers: ctx?.req?.headers?.cookie
      ? {
          cookie: ctx.req.headers.cookie,
        }
      : undefined,
  };
}
