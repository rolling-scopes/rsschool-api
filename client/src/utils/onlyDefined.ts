import { pickBy } from 'lodash';

export const onlyDefined = (data: object) => pickBy(data, val => val !== undefined && val !== '' && val !== null);
