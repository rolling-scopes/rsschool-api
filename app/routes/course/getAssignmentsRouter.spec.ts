import mockingoose from 'mockingoose';
import { createTestContext } from '../../utils';
import { getAssignments } from './getAssignmentsRouter';
import { STATES, connection } from 'mongoose';

describe('./getAssignmentsRouter.ts', () => {
    it('returns data', async () => {
        const ctx = createTestContext();
        mockingoose.assignments.toReturn([
            {
                assigmentRepo: 'https://github.com/',
                checkDate: 222,
                completeDate: 730260000,
                courseId: 'rs-course-2018-2',
                deadlineDate: 1537642920000.0,
                mentorComment: '',
                mentorId: '',
                score: 20,
                status: 'Assigned',
                studentComment: '',
                studentId: 'apalchys',
                taskId: '5b89b16999fa45417c962ae6',
                title: 'titleTask',
            },
        ]);
        connection.readyState = STATES.connected;
        await getAssignments(ctx);
        expect(ctx.status).toBe(200);
        const dateForSnapshot = {
            assigmentRepo: ctx.body.data[0].assigmentRepo,
            checkDate: ctx.body.data[0].checkDate,
            completeDate: ctx.body.data[0].completeDate,
            courseId: ctx.body.data[0].courseId,
            deadlineDate: ctx.body.data[0].deadlineDate,
            mentorComment: ctx.body.data[0].mentorComment,
            mentorId: ctx.body.data[0].mentorId,
            score: ctx.body.data[0].score,
            status: ctx.body.data[0].status,
            studentComment: ctx.body.data[0].studentComment,
            studentId: ctx.body.data[0].studentId,
            taskId: ctx.body.data[0].taskId,
            title: ctx.body.data[0].title,
        };
        expect(dateForSnapshot).toMatchSnapshot();
    });
});
