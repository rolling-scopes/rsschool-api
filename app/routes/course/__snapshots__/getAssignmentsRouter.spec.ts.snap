// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./getAssignmentsRouter.ts returns data 1`] = `
Object {
  "assigmentRepo": "https://github.com/",
  "checkDate": 222,
  "completeDate": 730260000,
  "courseId": "rs-course-2018-2",
  "deadlineDate": 1537642920000,
  "mentorComment": "",
  "mentorId": "",
  "score": 20,
  "status": "Assigned",
  "studentComment": "",
  "studentId": "apalchys",
  "taskId": "5b89b16999fa45417c962ae6",
  "title": "titleTask",
}
`;
